icon_validator/rules/workflow_validators/workflow_help_plugin_utilization_validator.py

```diff
@@ -0,0 +1,92 @@
+from icon_validator.rules.validator import KomandPluginValidator
+from icon_validator.exceptions import ValidationException
+
+import os
+import json
+import re
+
+
+class WorkflowHelpPluginUtilizationValidator(KomandPluginValidator):
+
+    @staticmethod
+    def extract_plugins_used(spec: object) -> list:
+        value = spec.directory
+        data = dict()
+        for file_name in os.listdir(value):
+            if file_name.endswith(".icon"):
+                with open(f"{value}/{file_name}") as json_file:
+                    try:
+                        data = json.load(json_file)
+                    except json.JSONDecodeError:
+                        raise ValidationException(
+                            "The .icon file is not in JSON format. Try exporting the .icon file again")
+        try:
+            plugins = data["kom"]["workflowVersions"][0]
+        except KeyError:
+            raise ValidationException("The .icon file is not formatted correctly. Try exporting the .icon file again")
+
+        plugin_list = list()
+        try:
+            for step in plugins["steps"]:
+                if "plugin" in plugins["steps"][step].keys():
+                    # The plugin_list must have a len of 1 or more for the rest of the code to work,
+                    # so the first item is just added
+                    if len(plugin_list):
+                        for index, d in enumerate(plugin_list):
+                            # Check if the Plugin + Version number are already in the list. If so add 1 to Count
+                            if d["Plugin"] == plugins["steps"][step]["plugin"]["name"] and d["Version"] == \
+                                    plugins["steps"][step]["plugin"]["slugVersion"]:
+                                plugin_list[index]["Count"] = plugin_list[index]["Count"] + 1
+                            # This was to check that the plugin was not in the list.
+                            # I don't think it's needed anymore but it works and I'm too scared to change it
+                            elif not any(d["Plugin"] == plugins["steps"][step]["plugin"]["name"] for d in
+                                         plugin_list) or not any(
+                                    d["Version"] == plugins["steps"][step]["plugin"]["slugVersion"] for d in
+                                    plugin_list):
+                                plugin_list.append({"Plugin": plugins["steps"][step]["plugin"]["name"],
+                                                    "Version": plugins["steps"][step]["plugin"]["slugVersion"],
+                                                    "Count": 1})
+                                # When a new plugin is added plugin_list's len goes up by one. This messes up this loop,
+                                # so a break is here to stop that
```

Comment:
I wouldn't do this...I'd make a copy of the list, and edit / analyze that as needed.